tutorials/thermal_camera.md

```diff
@@ -0,0 +1,268 @@
+        filename="ignition-gazebo-thermal-system"
+        name="ignition::gazebo::systems::Thermal">
+        <temperature>200.0</temperature>
+      </plugin>
+    </visual>
+  </link>
+</model>
+```
+
+Most of the code above is for the model - here's the key piece for temperature assignment:
+
+```xml
+<plugin
+  filename="ignition-gazebo-thermal-system"
+  name="ignition::gazebo::systems::Thermal">
+  <temperature>200.0</temperature>
+</plugin>
+```
+
+Once the thermal system plugin is specified, all we need to do is include the `<temperature>` element, which specifies the model's temperature in Kelvin.
+
+## Running an example:
+
+Now that we've discussed how a thermal camera and models with temperature can be specified, let's start an example world that uses the thermal camera.
+
+Run the following command:
+
+```
+ign gazebo -r thermal_camera.sdf
+```
+
+You should see something similar to this:
+
+@image html files/thermal_camera/thermal_camera_demo.png
+
+The window in the top-left corner shows the thermal camera's output.
+Taking a look at the [SDF file](https://github.com/ignitionrobotics/ign-gazebo/blob/c3391b1b664d1ec2b931d9a4ac757bde33b2a27b/examples/worlds/thermal_camera.sdf) for this example shows that the shapes were assigned the following temperatures:
+* sphere: 600 Kelvin
+* box: 200 Kelvin
+* cylinder: 400 Kelvin
+
+The temperature of objects in the view of the camera are normalized between 0 (black) and 255 (white) values.
+The object with the highest temperature in the camera's view will always have white color.
+If you move the object with the highest temperature out of the camera's view, the object with second highest temperature now becomes white.
+
+You can move the position of the shapes and/or camera around in the world to see the effect it has on the camera's output (the different "camera outputs" are how they are visualized in the GUI).
+An easy way to move objects in the world is by using `Transform Control`:
+
+@image html files/thermal_camera/thermal_camera_demo_2.png
+
+## Processing the thermal camera's output
+
+In the example above, the thermal camera publishes an [image message](https://github.com/ignitionrobotics/ign-msgs/blob/46a08597e6b6037adc98025cdc09dfbf0f4467a6/proto/ignition/msgs/image.proto) to the `/thermal_camera` topic whenever the camera has a new image.
+We can observe the contents of a single message by running the following command:
+
+```
+ign topic -e -n 1 -t /thermal_camera
+```
+
+Here are some important things to observe from the message output:
+* `width`: 320
+* `height`: 240
+* `pixel_format_type`: L_INT16 (16 bit integer)
+
+This means that the image is stored in the message as a serialized array of size 320x240 pixels.
+In order to do anything useful with the image stored in the message, we must convert the serialized array back to kelvin.
+Since the `pixel_format_type` is `L_INT16`, that means we need to cast the serialized array to an array of unsigned 16 bit integers.
```

Comment:
small suggestion about the use of`serialized array` - the users should be working with deserialized data already in their code in the topic callback. So maybe rephrase it to something like `raw image data` or `byte array`